### PR TITLE
Add negative http2 interop description file

### PR DESCRIPTION
**Pull request by ncteisen**
We have recently created a new set of tests to validate client behavior when it sees bad things from the server like GOAWAYs or RST_STREAMs. This documentation standardizes the tests.

### CODE REVIEW

File: doc/negative-http2-interop-test-descriptions.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+Server should accept these arguments:
+* --port=PORT
+  * The port the server will run on. For example, "8080"
+* --test_case=TESTCASE
+  * The name of the test case to execute. For example, "goaway"
+
+Client
+------
+
+Clients implement test cases that test certain functionally. Each client is
```
Comment (ericgribkoff) [inline doc/negative-http2-interop-test-descriptions.md]: functionality

File: doc/negative-http2-interop-test-descriptions.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+This test verifies that the client correctly responds to a goaway sent by the
+server. The client should handle the goaway by switching to a new stream without
+the user application having to do a thing.
+
+Client Procedure:
+ 1. Client sends two UnaryCall requests with:
+ 
+     ```
+    {
+      response_size: 1024
```
Comment (ericgribkoff) [inline doc/negative-http2-interop-test-descriptions.md]: Let's standardize these to use the same response_size and payload as the large_unary interop test case.

File: doc/negative-http2-interop-test-descriptions.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+Procedure:
+ 1. Client sends UnaryCall with:
+  ```
+  {
+    response_size: 1024
+    payload:{
+      body: 1024 bytes of zeros
+    }
+  }
```
Comment (ericgribkoff) [inline doc/negative-http2-interop-test-descriptions.md]: This code block is all showing on one line in Github's preview: https://github.com/ncteisen/grpc/blob/fcb2d1470664419a79692610f9aa60587f0aedc1/doc/negative-http2-interop-test-descriptions.md

It's fine in other markdown editor previews, not sure if it needs a newline before the start of the codeblock or something else is happening?

File: doc/negative-http2-interop-test-descriptions.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+Client asserts:
+* Call was not successful
+
+Server Procedure:
+  1. Server sends a RST_STREAM with error code 0 after sending all of the
+  data to the client.
+
+*Certain client languages allow the data to be accessed even though a RST_STREAM
+was encountered. Once all client languages behave this way, checks will be added on
+the incoming data.*
```
Comment (ericgribkoff) [inline doc/negative-http2-interop-test-descriptions.md]: I thought all languages could access the incoming data via the async API? Either way, we can resolve this later.

File: doc/negative-http2-interop-test-descriptions.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  {
+    response_size: 1024
+    payload:{
+      body: 1024 bytes of zeros
+    }
+  }
+  ```
+  
+Client asserts:
+* call was successful.
```
Comment (ericgribkoff) [inline doc/negative-http2-interop-test-descriptions.md]: nitpicking, but most of the other bullets are capitalized (and some don't have periods)

File: doc/negative-http2-interop-test-descriptions.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+Server Asserts:
+* Number of outstanding pings is 0 when the connection is lost
+
+### max_streams
+
+This test verifies that the client observes the MAX_CONCURRENT_STREAMS limit set by the server
+
+Client Procedure:
+  1. Client sends initial UnaryCall to allow the server to update its MAX_CONCURRENT_STREAMS settings.
+  2. Client asynchronously sends 15 UnaryCalls
```
Comment (ericgribkoff) [inline doc/negative-http2-interop-test-descriptions.md]: s/asynchronously/concurrently? Also, I thought we were doing 10 calls

File: doc/negative-http2-interop-test-descriptions.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+  }
+  ```
+
+Client asserts:
+* Call was not successful
+
+Server Procedure:
+  1. Server sends a RST_STREAM with error code 0 after sending headers to the client.
+  
+*At the moment the error code and message returned are not standardized throughout all
```
Comment (ericgribkoff) [inline doc/negative-http2-interop-test-descriptions.md]: Might be good to add link to https://github.com/grpc/grpc/issues/9142?

File: doc/negative-http2-interop-test-descriptions.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+
+Server Procedure:
+  1. Server sends a GOAWAY after receiving the first UnaryCall.
+
+Server asserts:
+* The second UnaryCall has a different stream_id than the first one.
+
+### rst_after_header
+
+This test verifies that the client fails "correctly" when the server sends a
```
Comment (ericgribkoff) [inline doc/negative-http2-interop-test-descriptions.md]: Not sure we need quotes around correctly here and below

File: doc/negative-http2-interop-test-descriptions.md [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+      response_size: 314159
+      payload:{
+        body: 271828 bytes of zeros
+      }
+    }
+    ```
+
+Client asserts:
+* Call was successful.
+* Response payload body is 1024 bytes in size.
```
Comment (ericgribkoff) [inline doc/negative-http2-interop-test-descriptions.md]: Payload body should be 314159 bytes in size

File: doc/negative-http2-interop-test-descriptions.md [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+      response_size: 314159
+      payload:{
+        body: 271828 bytes of zeros
+      }
+    }
+    ```
+  
+Client asserts:
+* call was successful.
+* response payload body is 1024 bytes in size.
```
Comment (ericgribkoff) [inline doc/negative-http2-interop-test-descriptions.md]: Same here

File: doc/negative-http2-interop-test-descriptions.md (hunk 7 above)
Comment (adelez) [inline doc/negative-http2-interop-test-descriptions.md]: Agree. Also error codes are different.

**Pull request merged**